packages/astro/src/types/public/config.ts

```diff
@@ -1864,6 +1927,36 @@ export interface ViteUserConfig extends OriginalViteUserConfig {
 		 */
 
 		responsiveImages?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.sessions
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 5.0.0
+		 * @see session
+		 * @description
+		 * Enables support for sessions in Astro. When enabled you can access the `Astro.session` object to read and write data that persists across requests. You can configure the session driver using the `session` option, or use the default provided by the adapter.
+		 *
+		 * ```js title=astro.config.mjs
+		 * {
+		 * 	experimental: {
+		 * 		sessions: true,
+		 * 	},
+		 * }
+		 * ```
+		 *
+		 * ```astro title=src/components/CartButton.astro
+		 * ---
+		 * const cart = await Astro.session.get('cart');
+		 * ---
+		 *
+		 * <a href="/checkout">🛒 {cart?.length ?? 0} items</a>
+		 *
+		 * ```
+		 *
```

Comment:
Link to RFC to learn more, maybe?

Reply:
Yeah, I'll do that once I have the stage 3 PR up

Reply:
Done!